cijoe/scripts/sysinfo_report.py

```diff
@@ -0,0 +1,405 @@
+            "0": "Unknown",
+            "1": "Other",
+            "2": "SIP",
+            "3": "DIP",
+            "4": "ZIP",
+            "5": "SOJ",
+            "6": "Proprietary",
+            "7": "SIMM",
+            "8": "DIMM",
+            "9": "TSOP",
+            "10": "PGA",
+            "11": "RIMM",
+            "12": "SODIMM",
+            "13": "SRIMM",
+            "14": "SMD",
+            "15": "SSMP",
+            "16": "QFP",
+            "17": "TQFP",
+            "18": "SOIC",
+            "19": "LCC",
+            "20": "PLCC",
+            "21": "BGA",
+            "22": "FPBGA",
+            "23": "LGA",
+            "24": "FB - DIMM",
+        }
+
+        for line in lines:
+            if SPEED in line:
+                speed = line.replace(SPEED, "").strip()[1:].strip()
+            elif FORMFACTOR in line:
+                formfactor = line.replace(FORMFACTOR, "").strip()[1:].strip()
+                formfactor = formfactor2name.get(formfactor, "Unknown")
+
+        err, state = cijoe.run(f'systeminfo | findstr /B /C:"{CAPACITY}"')
+        if err:
+            log.error(f"failed(systeminfo): err({err})")
+        capacity = (
+            state.output()
+            .strip()
+            .replace(CAPACITY, "")
+            .replace(".", "")
+            .strip()
+            .split()
+        )
+        if "MB" in capacity:
+            capacity = int(capacity[0]) // 1000
+        elif "GB" in capacity:
+            capacity = capacity[0]
+
+        return f"{capacity} GB {formfactor} {speed} MHz"
+
+    def get_drives() -> Dict[str, str]:
+        drives = {}
+
+        err, state = cijoe.run("get-disk | format-table Number, FriendlyName")
+        if err:
+            log.error(f"failed(get-disk): err({err})")
+        lines = lazy_read(state.output().strip().split("\n")[2:])
+
+        for line in lines:
+            match = re.search(r"^\s*(\d+) (.*)$", line)
+            if match is None:
+                continue
+            num, drive = match.groups(0)
+            drives[num] = drive
+
+        return drives
+
+    motherboard, cpu = get_motherboard_cpu()
+    memory = get_memory()
+    drives = get_drives()
+
+    return motherboard, cpu, memory, drives
+
+
+def main(args, cijoe, step):
+    operating_system = ""
+    kernel = ""
+    motherboard = ""
+    cpu = ""
+    memory = []
+    drives = {}
+
+    bios = ""
+    bios_version = ""
+
+    shell = cijoe.config.options.get("cijoe", {}).get("run", {}).get("shell", "sh")
+
+    if shell in ["cmd", "pwsh"]:
+        operating_system, kernel = read_os_windows(cijoe)
+        if shell == "pwsh":
+            motherboard, cpu, memory, drives = read_hardware_pwsh(cijoe)
+        else:
+            log.info("Reading hardware for Windows cmd not supported.")
+
+        # Get BIOS info
+        err, state = cijoe.run("wmic bios get biosversion")
+        if err:
+            log.error(f"failed(wmic bios get biosversion); err({err})")
```

Comment:
We should probably abort if we get here. Otherwise, we will crash immediately after.
The same goes for the following error paths